keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -142,6 +137,33 @@ class ResourceActuator(
       desired.await() to current.await()
     }
 
+  /**
+   * We want a specific status for specific types of vetos. This function publishes the
+   * right event based on which veto said no.
+   */
+  private fun publishVetoedEvent(response: VetoResponse, resource: Resource<*>) =
+    when {
+      response.vetoName == "ApplicationVeto" -> publisher.publishEvent(
+        ResourceActuationPaused(
+          resource.apiVersion,
+          resource.kind,
+          resource.id.value,
+          resource.spec.application,
+          response.message,
+          clock.instant()))
+      response.vetoName == "UnhappyVeto" -> {
+        // don't publish an event, we want the status to stay as "unhappy" for clarity
+      }
+      else -> publisher.publishEvent(
+        ResourceActuationVetoed(
```

Comment:
This isn't used yet (we only have two vetos), but will certainly be used in the future. Laying the plumbing now.